pkg/asset/installconfig/aws/metadata.go

```diff
@@ -106,25 +102,32 @@ func (m *Metadata) PrivateSubnets(ctx context.Context) (map[string]Subnet, error
 func (m *Metadata) PublicSubnets(ctx context.Context) (map[string]Subnet, error) {
 	err := m.populateSubnets(ctx)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "error retrieving Public Subnets")
 	}
-
 	return m.publicSubnets, nil
 }
 
+// VPC retrieves the VPC ID containing PublicSubnets and PrivateSubnets.
+func (m *Metadata) VPC(ctx context.Context) (string, error) {
+	err := m.populateSubnets(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "error retrieving VPC")
+	}
+	return m.vpc, nil
+}
+
 func (m *Metadata) populateSubnets(ctx context.Context) error {
-	if len(m.publicSubnets) > 0 || len(m.privateSubnets) > 0 {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if m.vpc != "" || len(m.privateSubnets) > 0 || len(m.publicSubnets) > 0 {
```

Comment:
@mtulio adding m.edgeSubnets to this tests causes the unit tests to fail. Looks like you specifically added a test to prevent that. I don't see why we can't check for len(m.edgeSubnets) > 0 here. Some special way EdgeSubnets are handled?

Reply:
@sadasu , sorry by missed that message. I am on it validating it to recall the scenario that drove me to create that unit blocking. Ideally, calling `subnets()` at once should populate all `*Subnets`, including `edge`.

I am [looking at the failed unit job on this PR history](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_installer/7070/pull-ci-openshift-installer-master-unit/1644019735538438144#1:build-log.txt%3A78-83), overall we required that edge subnets exist only when public or private have been provided, we should never accept the edge subnets only (as CP and regular workers must exist alongside LZ nodes) - this is a required defined on EP due AWS limitations of network resources (NLB/NGW, etc)

Would you mind sharing the current unit failure to help to dig in? I can see changes after the job I checked.

Reply:
also sharing on slack that I have installed successfully the cluster with Local Zones/edge + PHZ on this version
```
commit 1fced156cc5540a540a20ac208d79812ef72eae2 (HEAD -> pr_7070)
Author: Sandhya Dasu <sdasu@redhat.com>
Date:   Wed Apr 5 12:42:59 2023 -0400
```

install log:
```
$ OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="$RELEASE" $INSTALLER version
./openshift-install_pr7070 unreleased-master-7977-g1fced156cc5540a540a20ac208d79812ef72eae2-dirty
built from commit 1fced156cc5540a540a20ac208d79812ef72eae2
release image registry.ci.openshift.org/origin/release:4.13
release architecture amd64
(...)
INFO Time elapsed: 26m46s     
(...)                    
$ oc --kubeconfig $INSTALL_DIR/auth/kubeconfig get machines -n openshift-machine-api
NAME                                         PHASE     TYPE          REGION      ZONE               AGE
ocp-lz14-h9rmj-edge-us-east-1-nyc-1a-x9pkz   Running   c5d.2xlarge   us-east-1   us-east-1-nyc-1a   50m

```